forwarder.go

```diff
@@ -226,28 +232,64 @@ func buildExternalURL(newTalariaName, domain string) string {
  * Extracts IP address, certificate provider, and expiry details from HTTP request headers.
  * Calls the RI API with this information.
  */
-func updateResourceIpAddressAndCertificateInfo(req *http.Request, client *http.Client, resourceURL *url.URL) error {
+// Function to extract and decode X-WebPA-Convey header
+func extractAndDecodeXWebPA(req *http.Request) (UpdateResourceRequest, error) {
```

Comment:
Create a separate struct to decode the value of convey header and use that to populate UpdateResourceRequest.